OAuth: Allow assigning Server Admin (#54780)
grafana/grafana: docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth.md

role_attribute_path = contains(info.roles[*], 'admin') && 'Admin' || contains(info.roles[*], 'editor') && 'Editor' || 'Viewer'
```

#### Map server administrator privileges

> Available in Grafana v9.2 and later versions.

> If the role mapping expression evaluates to `GrafanaAdmin`, the user will be granted server administrator privileges.  

Reviewer: ```suggestion
If the role mapping expression evaluates to `GrafanaAdmin`, Grafana grants the user server administrator privileges.  
```

> This is useful if you want to grant server administrator privileges to a subset of users.  

Reviewer: See above suggestion

> The user will also be assigned the role of `Admin` of the default organization.

Reviewer: See suggestion above.

The setting `allow_assign_grafana_admin` under `[auth.generic_oauth]` must be set to `true` for this to work.  

> If the setting is set to `false`, the user will be assigned the role of `Admin` of the default organization, but not server administrator privileges.

Reviewer: ```suggestion
If the setting is set to `false`, the user is assigned the role of `Admin` of the default organization, but not server administrator privileges.
```
